runtime/vm/ContinuationHelpers.cpp

```diff
@@ -1239,6 +1239,16 @@ takeVirtualThreadListToUnblock(J9VMThread *currentThread)
 				case JAVA_LANG_VIRTUALTHREAD_TIMED_WAIT:
 					/* WAIT/TIMED_WAIT can only be added to blocked list if they have been notified. */
 					Assert_VM_true(J9VMJAVALANGVIRTUALTHREAD_NOTIFIED(currentThread, current->vthread));
+					/* The transition to BLOCKED may have been missed if vthread was in a WAITING state while notified. */
+					if (MM_ObjectAccessBarrierAPI(currentThread).inlineMixedObjectCompareAndSwapU32(
+									currentThread,
+									current->vthread,
+									J9VMJAVALANGVIRTUALTHREAD_STATE_OFFSET(currentThread),
+									state,
+									JAVA_LANG_VIRTUALTHREAD_BLOCKED)
+					) {
+						continue;
```

Comment:
Im not sure if continue is the right action here, if a thread has been notified, but wasn't set to blocked due to timing, then after successful update to the state, it should be added to the unblocked list to be processed by the unblocker. so will need to handle it similar to the `JAVA_LANG_VIRTUALTHREAD_BLOCKED` case below.